### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by AnsarIbrahim**
### Project requirements

- [x] **Design:**
     - I should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)
     - I added Some styles to improve and in My Idea

- [x] **Interactions:**

 - [x] **Splash screen**
     - A simple page with the name of the Budget app and links to the sign-up and log-in links in there

- [x] **Sign up and login pages**
    - The user should be able to register in the app with full name, email, and password (all mandatory).
    - The user can log into the app using email and password.
    - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [x] **Home page (categories page)**
   - When the user logs in, they are presented with the categories page.
   - For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
   - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
   - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [x] **Transactions page**
     - For a given category, the list of transactions is presented, ordered by the most recent.
     - At the top of the page, the user can see the total amount for the category (sum of all of the amounts of the transactions in that category).
     - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
     - When the user clicks on the "Back" button (<), the user navigates to the home page.

- [x] **"Add a new category" page**
     - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
     - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
     - When the user clicks on the "Back" button (<), the user navigates to the home page.

- [x] **"Add a new transaction" page**
     - The user fills out a form to create a new transaction with:
          - name (mandatory)
          - amount (mandatory)
          - categories (mandatory at least one)
     - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
     - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- [x] **Testing requirements**
    - Create unit and integration tests for all the most important components of My RoR application.

- [x] **Technical requirements**
    - I should use Postgres as your database.
    - I should use a device for authentication.
    - I should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
    - I can use a view template engine of your choice (.erb, .slim, .haml).
    - The project should be deployed and accessible online.
    - My database schema should reflect the following structure:

- [x] **Project documentation**
   - I recorded a video from the loom and I attached it to the Readme file
   - The Project Deploy link is also Attached to the readme file

This pull request is ready for review and merging into the main codebase.

Thank you for your time and attention to this pull request!

Regards,

[Ansar Ibrahim](@AnsarIbrahim)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "githubPullRequests.ignoredPullRequestBranches": [
+        "dev"
+    ]
+}
```
Comment (mukhammadrizooff) [inline .vscode/settings.json]: - [ ] [REQUIRED] Good job so far! I appreciate what you have done your design however small parts of your design should follow the [Behance](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) provided on `requirement`. There your `Log In & Sign Up` page's buttons placed wrongly and implment background color `white` as [Behance](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) 🙏🏼 

image

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "githubPullRequests.ignoredPullRequestBranches": [
```
Comment (mukhammadrizooff) [inline .vscode/settings.json]: - [ ] [REQUIRED] Good job so far! I appreciate what you have done your design however small parts of your design should follow the [Behance](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) provided on `requirement`. There `home` page part `"Add new category"` button should locate below of the page 🙏🏼 

image

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (mukhammadrizooff) [inline .vscode/settings.json]: - [ ] [REQUIRED] Good job so far! On my side few `specs` failed when they run, please fix them if your side everything is ok please show screenshot to next `Code Reviewer` 🙏🏼 

image

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @AnsarIbrahim, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Clean work ✔️ 
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [ ] [OPTIONAL] Good job so far! It would be more proffessional if you submit your project as `main from dev`✔️ not a `dev from any_branch`❌, because that kind of situation some of requests addressed not clear 🙏🏼 

image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AnsarIbrahim** [comment]
Change UI and here I attach a screenshot of all that the last reviewer mentioned it

Screenshot 2023-10-19 175238

Screenshot 2023-10-19 175223

Screenshot 2023-10-19 175628

**Review by joel-kalema — APPROVED** [review]
Hi @AnsarIbrahim 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights
The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**